File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-0.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int l = 0, r = 0, res = Integer.MAX_VALUE, sum = 0;

        while(r < nums.length){
            sum += nums[r];
            while(sum >= target){
                res = Math.min(res, r-l+1);
                sum -= nums[l];
                l++;
            }
            r++;
        }
        return res == Integer.MAX_VALUE ? 0 : res;
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        mapp = {}

        for i in range(len(nums)):
            val = target - nums[i]
            if(val in mapp):
                return [mapp[val], i]
            mapp[nums[i]] = i
        
        return []